taichi/program/state_flow_graph.cpp

```diff
@@ -127,13 +127,19 @@ void StateFlowGraph::insert_task(const TaskLaunchRecord &rec) {
                       input_state);
   }
   for (auto output_state : node->meta->output_states) {
-    latest_state_owner_[output_state] = node.get();
     if (latest_state_readers_.find(output_state) ==
         latest_state_readers_.end()) {
-      latest_state_readers_[output_state].insert(initial_node_);
+      if (latest_state_owner_.find(output_state) != latest_state_owner_.end()) {
+        // insert a WAW dependency edge
+        insert_state_flow(latest_state_owner_[output_state], node.get(),
```

Comment:
WDYT if we change the name a bit? `insert_state_flow()` sounds like this is a... state flow (rather than a dependency flow). Maybe just `insert_edge`, `connect`, or something..?